modules/dists/SparseBlockDist.chpl

```diff
@@ -117,6 +119,102 @@ class SparseBlockDom: BaseSparseDom {
     dsiAdd((ind,));
   }
 
+  proc dsiBulkAdd(inds: [] index(rank, idxType), isSorted=false, isUnique=false,
+      preserveInds=true){
+  
+    if !isSorted && preserveInds {
+      var _inds = inds;
+      bulkAdd_help(_inds, isSorted, isUnique); 
+    }
+    else {
+      bulkAdd_help(inds, isSorted, isUnique);
+    }
+  }
+
+  // Tried to put this record in the function and the if statement, but got a
+  // segfault from the compiler.
+  record TargetLocaleComparator {
+
+    // I wish I could return tuple from key function. There are checks to see
```

Comment:
I don't think it's really necessary for this PR because at the moment we're comparison sorting anyway... but agree it would be a nice improvement for Sort.